### ISSUE TITLE
Cannot log in from Angular client after migration from 3.3.2 to 4.3.0

### ISSUE THREAD

**Issue by alexandis** [abpframework/abp#8779]
**ABP 4.3.0** after migration at backend
**Angular** frontend (not ready to be migrated to **4.3.0** yet! it's **3.3.2**)
**EF Core**

My backend code is finally successfully migrated to **ABP 4.3.0** and I can log in on **IdentityServer** page. Bug when I try to log in from my **Angular** app, getting this:

> 2021-04-25 10:11:55.154 +03:00 [DBG] Start authorize request protocol validation
> 2021-04-25 10:11:55.168 +03:00 [DBG] client configuration validation for client XXX_App succeeded.
> 2021-04-25 10:11:55.168 +03:00 [DBG] Checking for PKCE parameters
> 2021-04-25 10:11:55.741 +03:00 [ERR] Scope XXX not found in store.
> 2021-04-25 10:11:55.741 +03:00 [ERR] Request validation failed

From what I have found out, **IdentityServer** functionality has been changed lately and now I need to have **ApiScope** filled in (and store them in memory to be checked by IdentityServer?)
Do I get it right my problem is related to empty tables **IDENTITYSERVERAPISCOPECLAIMS** and **IDENTITYSERVERAPISCOPEPROPERTIES** or some additional setting is required in **IdentityServer** backend part?
If so, after migration these tables appeared to be empty. Neither I can fill them in, since at first I need to log in into **Angular** app as an _admin_. What is correct way to resolve the issue? Do I need to create a separate _Web_ app to log in and set up scopes there?

![image](https://user-images.githubusercontent.com/6149843/115986439-d6b9d300-a5b8-11eb-9cb4-330a349dee45.png)

P.S. **IDENTITYSERVERAPISCOPES** is filled with all scope data correctly.

**Comment by maliming** [comment]
Check the **Migration-Guides** before the upgrade.

https://docs.abp.io/en/abp/latest/Migration-Guides/Index

**Comment by alexandis** [comment]
> Check the **Migration-Guides** before the upgrade.
> 
> https://docs.abp.io/en/abp/latest/Migration-Guides/Index

The matter is i already DID the migration.

UPDATE: I run **DbMigrator** and seeded data - it took a while, but there was no errors. Afterwards, I've updated all my custom **Angular** _libraries_ and main project with the latest **ABP** modules. However, I still cannot log in. On backend side everything seems to be successful, but **Bearer** is not written and I am redirected back to **Angular** login page having only this:

![image](https://user-images.githubusercontent.com/6149843/116169673-ed257300-a70d-11eb-936e-62dd7be00a3f.png)

Last thing from **IdentityServer** is:

> [04:03:14 DBG] Token request success.
> [04:03:14 INF] Request finished HTTP/2 POST https://localhost:44357/connect/token application/x-www-form-urlencoded 259 - 200 - application/json;+charset=UTF-8 6992.9730ms

Maybe some required data still missing in IDENTITYSERVER_XXXXXXX tables after migration??

I have also noticed, that after logging in at **IdentityServer** loginbox I'm getting `refresh_token`, `id_token` and some other data for very short period of time (even cannot make a note at all data) and in less than a second these fields are removed and only a couple of fields above are remained in _Local storage_.

**Comment by alexandis** [comment]
UPDATE: using Angular test app, generated via ABP Suite 4.3.0 (but consuming my Environment data), I am having the same issue with login.

**Comment by maliming** [comment]
> but consuming my Environment data

The application port is randomly generated, so there will be problems. You can try to use the template migration to create a database to test.

**Comment by alexandis** [comment]
> > but consuming my Environment data
> 
> The application port is randomly generated, so there will be problems. You can try to use the template migration to create a database to test.

What do you mean? I use fully my backend - where all ports are fixed. I use only Angular app on port 4200 to connect to my backend...

**Comment by maliming** [comment]
I suggest you create a new project and a new database to test and compare the code and data.

**Comment by alexandis** [comment]
> I suggest you create a new project and a new database to test and compare the code and data.

It's not easy and obvious to compare side-by-side two solutions, because we have a lot of custom logic and it would take a while to set up local Oracle DB for me.

So I have created a _multi-tiered_ app from **ABP** _template_ and adopted this **Web** app for our solution. I tried to look what **IdentityServer** and **Web** projects in the generated template look like (appsettings, Module configuring). There is nothing especial there, besides the fact it uses Twitter/Google/OpenId authentication. I have replaced all this with our existing code:

        private void ConfigureAuthentication(ServiceConfigurationContext context, IConfiguration configuration)
        {
            context.Services.AddAuthentication()
                .AddJwtBearer(options =>
                {
                    options.Authority = configuration["AuthServer:Authority"];
                    options.RequireHttpsMetadata =  Convert.ToBoolean(configuration["AuthServer:RequireHttpsMetadata"]);;
                    options.Audience  = "CentralTools";
                    options.BackchannelHttpHandler = new HttpClientHandler
                    {
                        //TODO: use valid certificate in future and change the logic
                        ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator
                    };
                });

            context.Services.AddAbpIdentity().AddDefaultTokenProviders();
        }

It does not work either: seems like after pressing Login I'm not redirected to our **IdentityServer** (even though it's indicated properly in appsettings.json):

![image](https://user-images.githubusercontent.com/6149843/116273850-c9ecd900-a78a-11eb-84b2-d59e3411b6e6.png)

![image](https://user-images.githubusercontent.com/6149843/116273684-a7f35680-a78a-11eb-8a2f-7dad2cd417d2.png)

So the logic of overriden ProfileService is not executed:

![image](https://user-images.githubusercontent.com/6149843/116270883-2d293c00-a788-11eb-8caf-be8594b07053.png)

![image](https://user-images.githubusercontent.com/6149843/116271034-4c27ce00-a788-11eb-9074-469085f0be43.png)

It used to work properly with ABP 3.3.2 and it IS called when accessing from Angular app.

What am I missing? I am not allowed to share the code, it's a commercial project.

Could it have something to do with wrong DefaultPassPhrase, for instance? What is this about?? I can't find anything on ABP site about them.
